indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,50 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable, AffiliateInfoTable, PersistentCacheKeys, PersistentCacheFromDatabase,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+import config from '../config';
+
+const defaultLastUpdateTime: string = '2023-10-26T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  try {
+    const start = Date.now();
+    const persistentCacheEntry: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+      .findById(PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME);
+
+    if (!persistentCacheEntry) {
+      logger.info({
+        at: 'update-affiliate-info#runTask',
+        message: `No previous ${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME} found in persistent cache table. Will use default value: ${defaultLastUpdateTime}`,
```

Comment:
this is different from `update-wallet-total-volume`, because the first instance of fills is 10/26/23, but since this task is updating affiliate data for affiliate addresses, you can just use 9/19/23, because there will never be any affiliate data before today

Reply:
changed